src/web/components/Core/EnvironmentBanner.tsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react';
+
+import { EnvironmentVariable, GetEnvironmentVariables } from '../../services/environmentVariables';
+import { ApiError } from '../../utils/apiError';
+import { useAsyncError } from '../../utils/errorHandler';
+import { Banner } from './Banner';
+
+function EnvironmentBanner() {
+  const [environment, setEnvironment] = useState<EnvironmentVariable>({
+    baseUrl: '',
+    isDevelopment: false,
+  });
+  const throwError = useAsyncError();
+
+  useEffect(() => {
```

Comment:
I think this is OK, this part of the component tree should be rendered rarely so I don't think we'll end up fetching this often. Normally I'd suggest something like this should be provided via context rather than fetched, but in this case it looks fine 🙂

Reply:
yeah I considered using context but came to the same conclusion. thanks!